public/app/plugins/panel/heatmap/tooltip/utils.ts

```diff
@@ -45,7 +45,8 @@ export const formatMilliseconds = (milliseconds: number) => {
     }
   }
 
-  const unitString = value === 1 ? unit : unit + 's';
+  const plural = value !== 1 && !['ms', 's', 'm', 'h'].includes(unit);
```

Comment:
maybe pull this array out/above this function and do something like `const noPluralize = new Set(['ms', 's', 'm', 'h'])`. then inside the function `&& !noPluralize.has(unit)`. that way it doesnt recreate the static array on every hover. and `Set.has()` is faster than `Array.includes()` ;)